src/Tablebot/Util/Random.hs

```diff
@@ -0,0 +1,48 @@
+-- |
+-- Module      : Tablebot.Util.Random
+-- Description : A plugin for random generation.
+-- Copyright   : (c) Amelie WD 2021
+-- License     : MIT
+-- Maintainer  : tablebot@ameliewd.com
+-- Stability   : experimental
+-- Portability : POSIX
+--
+-- A collection of utility functions for generating randomness.
+module Tablebot.Util.Random (chooseOne, chooseOneWithDefault, chooseOneWeighted, chooseOneWeightedWithDefault) where
+
+import Data.List
+import Data.Maybe
+import System.Random (randomRIO)
+import Tablebot.Util.Error
+
+-- | @chooseOne@ chooses a single random element from a given list with uniform
+-- distribution.
+chooseOne :: [a] -> IO (Either Error a)
+chooseOne [] = return $ Left "Cannot choose from empty list."
+chooseOne xs = Right . (xs !!) <$> randomRIO (0, length xs - 1 :: Int)
```

Comment:
I've made my opinions on `!!` known - and would support using `chooseOneWeighted` under the hood even if it has those extra checks

Reply:
I don't like it, but agree that this is likely the best way to do it so I have no problems.